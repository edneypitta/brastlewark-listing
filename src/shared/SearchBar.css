.wrapper {
  margin-bottom: 20px;
  height: 10vh;
}

.wrapper input {
  width: 100%;
  border: none;
  border-bottom: 1px solid #fff;
  background: transparent;
  height: 40px;
  color: #fff;
  outline: none;
  padding: 0 10px;
  font-weight: bold;
}
