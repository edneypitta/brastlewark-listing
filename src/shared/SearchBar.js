import React from 'react'
import PropTypes from 'prop-types'
import './SearchBar.css'

const SearchBar = ({ value, onChange }) => 
  <div className="wrapper">
    <input
      type="text"
      placeholder="Search"
      value={value}
      onChange={onChange}/>
  </div>

SearchBar.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

SearchBar.defaultProps = {
  value: ''
}

export default SearchBar