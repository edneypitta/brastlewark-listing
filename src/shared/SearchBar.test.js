import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
  it('renders without crashing', () => {
    const component = shallow(<SearchBar value="" onChange={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })
})
