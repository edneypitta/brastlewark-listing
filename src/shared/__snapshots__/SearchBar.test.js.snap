// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders without crashing 1`] = `
<div
  className="wrapper"
>
  <input
    onChange={[MockFunction]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
